aten/src/ATen/templates/Tensor.h

```diff
@@ -288,7 +288,8 @@ class CAFFE2_API Tensor {
   T * data_ptr() const;
 
   template<typename T>
-  T * data() const {
+  C10_DEPRECATED_MESSAGE("tensor.data_ptr<T>() is deprecated. Please use tensor.data_ptr<T>() instead.")
```

Comment:
nit: probably slightly better to say `Tensor.data_ptr` rather than `tensor.data_ptr`.